addons/account/models/chart_template.py

```diff
@@ -528,13 +528,13 @@ def deref_values(values, model):
                         if model != self.env['res.company']:
                             _logger.warning("Failed when trying to recover %s for field=%s", value, field)
                             failed_fields.append(fname)
-
-                        # We can't find the record referenced in the chart template in our database.
-                        # This might happen when we're creating a branch and the parent company has deleted the
-                        # referenced record and replaced it with something else.
-                        #
-                        # In this case, we try looking for the record already set on the company or its root.
-                        values[fname] = self.env.company[fname] or self.env.company.parent_ids[0][fname] or False
+                        else:
```

Comment:
It would be more logical the other way around:
```python
if model._name == 'res.company':
    # Try a fallback on the company when reloading/loading on a branch
    values[fname] = self.env.company[fname] or self.env.company.root_id[fname] or False
else:
    _logger.warning("Failed when trying to recover %s for field=%s", value, field)
    failed_fields.append(fname)
```

Also, I'm pretty sure that `values[fname] = self.env.company[fname]` can fail because `write` expects an id and not a recordset :confused: